telex/telex.go

```diff
@@ -0,0 +1,219 @@
+// package telex executes process and streams process output to clients
+package telex
+
+import (
+	"context"
+	"errors"
+	"os/exec"
+	"sync"
+
+	"github.com/google/shlex"
+	"github.com/google/uuid"
+
+	"github.com/qrdl/telex/spooler"
+)
+
+// Process status codes
+const (
+	StatusInvalid = iota
+	StatusRunning
+	StatusCompleted
+	StatusFailed
+	StatusKilled
+)
+
+// Output stream types
+const (
+	_ = iota
+	StreamStdout
+	StreamStderr
+)
+
+const invalidHandle = ""
+
+// cache of all executed and currently executing processes
+type process struct {
+	sync.Mutex
+	status      int
+	owner       string
+	exitcode    int
+	stdout      *spooler.Spooler
+	stderr      *spooler.Spooler
+	killChannel chan bool
+}
+
+type jobLog struct {
+	sync.RWMutex // mutex to allow safely adding new records to runLog
+	log   map[uuid.UUID]*process
+}
+
+var jobs = jobLog{log: map[uuid.UUID]*process{}}
+
+// StartProcess starts execution of provided command, returns process handle or error.
+func StartProcess(cmdLine string, user string) (string, error) {
+	params, err := shlex.Split(cmdLine)
+	if err != nil {
+		return invalidHandle, err
+	}
+
+	var proc *exec.Cmd
+	if len(params) > 1 {
+		proc = exec.Command(params[0], params[1:]...)
+	} else {
+		proc = exec.Command(params[0])
+	}
```

Comment:
this should work without the check
```suggestion
	proc = exec.Command(params[0], params[1:]...)
```

Reply:
right, `foo[len(foo):]` is legal. Will change it

Reply:
Changed